File: test/language/enumeration_types/java/enumeration_types/enum_defined_by_constant/EnumDefinedByConstantTest.java
package enumeration_types.enum_defined_by_constant;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

public class EnumDefinedByConstantTest
{
    @Test
    public void colors()
    {
        assertEquals(1, WHITE_COLOR.WHITE_COLOR);
        assertEquals(WHITE_COLOR.WHITE_COLOR, Colors.WHITE.getValue());
        assertEquals(Colors.WHITE.getValue() + 1, Colors.BLACK.getValue());
    }
}

File: test/language/enumeration_types/cpp98/EnumDefinedByConstantTest.cpp
#include "gtest/gtest.h"

#include "enumeration_types/enum_defined_by_constant/Colors.h"
#include "enumeration_types/enum_defined_by_constant/WHITE_COLOR.h"

namespace enumeration_types
{
namespace enum_defined_by_constant
{

class EnumDefinedByConstant : public ::testing::Test
{
};

TEST_F(EnumDefinedByConstant, lightColor)
{
    ASSERT_EQ(1, WHITE_COLOR);
    ASSERT_EQ(WHITE_COLOR, Colors::White);
    ASSERT_EQ(Colors::White + 1, Colors::Black);
}

} // namespace enum_defined_by_constant
} // namespace enumeration_types
